src/core/Elsa.Abstractions/ActivityResults/SuspendResult.cs

```diff
@@ -1,16 +1,24 @@
+using Elsa.Events;
 using Elsa.Models;
 using Elsa.Services.Models;
+using MediatR;
+using Microsoft.Extensions.DependencyInjection;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace Elsa.ActivityResults
 {
     public class SuspendResult : ActivityExecutionResult
     {
-        protected override void Execute(ActivityExecutionContext activityExecutionContext)
+        public override async ValueTask ExecuteAsync(ActivityExecutionContext activityExecutionContext, CancellationToken cancellationToken)
         {
             var activityDefinition = activityExecutionContext.ActivityBlueprint;
             var blockingActivity = new BlockingActivity(activityDefinition.Id, activityDefinition.Type);
             activityExecutionContext.WorkflowExecutionContext.WorkflowInstance.BlockingActivities.Add(blockingActivity);
             activityExecutionContext.WorkflowExecutionContext.Suspend();
+
+            var mediator = activityExecutionContext.ServiceProvider.GetRequiredService<IMediator>();
+            await mediator.Publish(new BlockingActivityRemoved(activityExecutionContext.WorkflowExecutionContext, blockingActivity));
```

Comment:
Shouldn't this be `BlockingActivityRemoved` since we added a blocking activity?

Reply:
fixed @sfmskywalker